src/modules/person/api/person.controller.ts

```diff
@@ -74,32 +72,13 @@ export class PersonController {
     @ApiForbiddenResponse({ description: 'Insufficient permissions to create the person.' })
     @ApiInternalServerErrorResponse({ description: 'Internal server error while creating the person.' })
     public async createPerson(@Body() params: CreatePersonBodyParams): Promise<PersonendatensatzResponse> {
-        const person: Person<false> = Person.createNew(
-            params.name.familienname,
-            params.name.vorname,
-            params.referrer,
-            params.stammorganisation,
-            params.name.initialenfamilienname,
-            params.name.initialenvorname,
-            params.name.rufname,
-            params.name.titel,
-            params.name.anrede,
-            params.name.namenspraefix,
-            params.name.namenssuffix,
-            params.name.sortierindex,
-            params.geburt?.datum,
-            params.geburt?.geburtsort,
-            params.geschlecht,
-            params.lokalisierung,
-            params.vertrauensstufe,
-            params.auskunftssperre,
-        );
+        const person: Person<false> = await this.personFactory.createNew({
+            vorname: params.name.vorname,
+            familienname: params.name.familienname,
+            ...params,
+        });
```

Comment:
@kristoff-kiefer Das sieht für mich nicht richtig aus.

Das Objekt welches du an die createNew Methode gibst hat diese Struktur:
{
vorname:x,
familienname:x,
email:x,
referrer:x,
stammorganisation:x,
name:{
familienname:x,
vorname:x,
.....
},
....


Das erwatete PersonCreationParams-DTO der createNew Methode hat allerdigs eine ganz andere Struktur, welche nur top level props vorsieht. Oder habe ich hier einen Denkfehler?

Reply:
In Call geklärt